components/locid/src/helpers.rs

```diff
@@ -44,12 +45,88 @@ impl<T> ShortVec<T> {
         }
     }
 
+    #[inline]
+    pub fn as_mut_slice(&mut self) -> &mut [T] {
+        match self {
+            ShortVec::Empty => &mut [],
+            ShortVec::Single(v) => core::slice::from_mut(v),
+            ShortVec::Multi(v) => v.as_mut_slice(),
+        }
+    }
+
+    #[inline]
     pub const fn single(&self) -> Option<&T> {
         match self {
             ShortVec::Single(v) => Some(v),
             _ => None,
         }
     }
+
+    #[inline]
+    pub fn len(&self) -> usize {
+        match self {
+            ShortVec::Empty => 0,
+            ShortVec::Single(_) => 1,
+            ShortVec::Multi(ref v) => v.len(),
+        }
+    }
+
+    #[inline]
+    pub fn insert(&mut self, index: usize, elt: T) {
+        assert!(
+            index <= self.len(),
+            "insertion index (is {}) should be <= len (is {})",
+            index,
+            self.len()
+        );
+
+        *self = match core::mem::replace(self, ShortVec::Empty) {
+            ShortVec::Empty => ShortVec::Single(elt),
+            ShortVec::Single(item) => {
+                let mut items = vec![item];
+                items.insert(index, elt);
+                ShortVec::Multi(items)
+            }
+            ShortVec::Multi(mut items) => {
+                items.insert(index, elt);
+                ShortVec::Multi(items)
+            }
+        }
+    }
+
+    #[inline]
+    pub fn remove(&mut self, index: usize) -> T {
+        assert!(
+            index < self.len(),
+            "removal index (is {}) should be < len (is {})",
+            index,
+            self.len()
+        );
+
+        let (replaced, maybe_removed_item) = match core::mem::replace(self, ShortVec::Empty) {
+            ShortVec::Empty => (ShortVec::Empty, None),
```

Comment:
Suggestion: since the vec cannot be empty, replace this with `unreachable!()` and make `maybe_removed_item` be non-optional